EIPS/eip-micropayments_standard_for_nfts_and_multi_tokens.md

```diff
@@ -0,0 +1,329 @@
+---
+eip: 4393
+title: Micropayments Standard for NFTs and Multi Tokens
+description: A standard interface for tip tokens that allows tipping to holders of NFTs and multi tokens
+author: Jules Lai (@julesl23)
+discussions-to: https://ethereum-magicians.org/t/eip-proposal-micropayments-standard-for-nfts-and-multi-tokens/7366
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-10-24
+requires: 165, 721, 1155
+---
+
+## Abstract
+
+For the purpose of this proposal, a micropayment is termed as a financial transaction that involves usually a small sum of money, called "tips", that are sent for specific [ERC-721](./eip-721.md) NFTs and [ERC-1155](./eip-1155.md) multi tokens as rewards to their holders. A holder (also referred to as controller) is used here as a more generic term for owner in a semantic sense, as NFTs may represent non-digital assets, such as services etc.
+
+This standard proposal outlines a generalised way to allow tipping via implementation of an `ITipToken` interface. The interface is intentionally kept to a minimum in order to allow for maximum use cases.
+
+A user first deposits a compatible [ERC-20](./eip-20.md) to the tip token contract that is then held in escrow, in exchange for tip tokens. These tip tokens can then be sent by the user to NFTs and multi-tokens (that have been approved by the tip token contract for tipping) to be redeemed for the original ERC20 deposits on withdrawal by the holders as rewards.
+
+Some use cases include:
+
+- In game purchases and other virtual goods
+
+- Tipping messages, posts, music and video content
+
+- Donations/crowdfunding
+
+- Distribution of royalties
+
+- Pay per click advertising
+
+- Incentivising use of services
+
+- Reward cards and coupons
+
+These can all leverage the security, immediacy and transparency of blockchain.
+
+## Motivation
+
+A cheap way to send tips to any type of NFT or multi token. This can be achieved by gas optimising the tip token contract and sending the tips in batches using the `tipBatch` function from the interface.
+
+To make it easy to implement into dapps a tipping service to reward the NFT and multi token holders. Allows for fairer distribution of revenue back to NFT holders from the user community.
+
+To make the interface as minimal as possible in order to allow adoption into many different use cases.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+Smart contracts implementing the ERC-xxxx standard MUST implement all of the functions in the ERC-xxxx interface. MUST also emit the events specified in the interface so that a complete state of the tip token contract can be derived from the events emitted alone.
```

Comment:
Need to replace EIP-xxxx -- usually prefer to not refer to the EIP itself by its own number. Instead say "this EIP" or a something similar.

Reply:
Done